### ISSUE TITLE
support convert negative integer to bitarray

### ISSUE THREAD

**Issue by apurance** [ilanschnell/bitarray#85]
function int2ba() do **NOT** support negative integer
as in bitarray/util.py line:164-165
> if i < 0:
        raise ValueError("non-negative integer expected")

so I suggest that:

```python
#roughly
def int2ba(n, width):
    ba = bitarray()
    for i in range(width):
        ba.append((n >> i) & 1)
    return ba
```
whether the n is positive or negative, the complement specified by width can be output correctly

for example:
```shell
>>> int2ba(127, 8)
bitarray('11111110')
>>> int2ba(-128, 8)
bitarray('00000001')
```

when width == 8
it seems that the input interval is -128 to 127, and output complement, while the original function requires the input interval is 0 to 255, and output unsigned true form
in fact, when the input is in the interval [-128, 127] and interval [0, 255], this function will output complement and unsigned true form respectively
for example:
```shell
>>> int2ba(-128, 8)
bitarray('00000001')
>>> int2ba(128, 8)
bitarray('00000001')
```
for bitarray('00000001'), it is interpreted as -128 when it is used as complement, meanwhile it is interpreted as 128 when it is used as unsigned true form

**Comment by apurance** [comment]
when width == 8
it seems that the input interval is -128 to 127, and output complement, while the original function requires the input interval is 0 to 255, and output unsigned true form
in fact, when the input is in the interval [-128, 127] and interval [0, 255], this function will output complement and unsigned true form respectively
for example:
```shell
>>> int2ba(-128, 8)
bitarray('00000001')
>>> int2ba(128, 8)
bitarray('00000001')
```
for bitarray('00000001'), it is interpreted as -128 when it is used as complement, meanwhile it is interpreted as 128 when it is used as unsigned true form

**Comment by ilanschnell** [comment]
Thank you for pointing this out.  I think it would be best to add an optional `signed` keyword argument to `int2ba`, which determines whether two's complement is used to represent the integer (basically does the same as in `int.to_bytes()`), and defaults to `False`.  So we would have:
```
>>> int2ba(-1, 8)
OverflowError: can't convert negative int to unsigned
>>> int2ba(-1, 4, signed=True)
bitarray('1111')
```
Using `signed=True` will only make sense when length is given.  The range of possible integers will (as it already does now) on the length, e.g. for length 8: `-128 .. 127` and for length 5: `-16 .. 15`.

In addition, it would make sense to add the signed keyword argument to `ba2int`.

**Comment by apurance** [comment]
yes, for `ba2int`, I suggest that:
```python
def ba2int(ba, signed=False):
    n = 0
    l_1 = len(ba) - 1
    for i in range(l_1):
        n += ba[i] << i
    if signed:
        n -= ba[l_1] << l_1
    else:
        n += ba[l_1] << l_1
    return n
```
THANKS **bitarray**
comparing with storing bits directly on list, bitarray improves efficiency by at least 36%

**Comment by ilanschnell** [comment]
@apurance I've implemented this new feature, and it will be part of the next release.

**Comment by ilanschnell** [comment]
I just released 1.5.1 with signed integer support in `util.ba2int()` and `util.int2ba()`.
